src/dbnode/generated/thrift/rpc.thrift

```diff
@@ -312,6 +314,41 @@ struct HealthResult {
 	2: required string status
 }
 
+enum AggregateQueryType {
+	AGGREGATE_BY_TAG_NAME,
+	AGGREGATE_BY_TAG_NAME_VALUE,
+}
+
+// AggregateQueryRequest comes from our desire to aggregate on incoming data.
+// We currently only support retrieval of facets, but could extend this based on
+// requirements in the future. Currently, the two predominant use-cases are:
+// (1) Given a filter query (optionally), return all known tag keys matching this restriction
+// (2) Given a filter query (optionally), return all know tag key+values matching this restriction
```

Comment:
Should this take in an optional result filter on keys? As an example, if I'm doing a graphite query for say, `foo.bar.*`, I only want to list the values matching the restriction for `__g2__`, rather than for any tags that come after. For graphite we can do something silly like add `__g3__ !=~ ".*"` to soft-restrict this to the desired output, but that doesn't really translate well to arbitrary tag/values

i.e. with `foo.bar.*` as a query, the expected results for some metrics would be:
```
foo.bar.biz -> biz
foo.bar.qux -> qux
foo.bar.a.b -> a
```

(sorry for using mostly graphite here, it's easier than writing out tags explicitly for the example :p)

Reply:
I like it. I'll add something equivalent to `resultFilter []string` where empty means no filtering